src/main/java/io/jenkins/plugins/audit/listeners/UserLogListener.java

```diff
@@ -0,0 +1,154 @@
+
+import io.jenkins.plugins.audit.event.Login;
+import io.jenkins.plugins.audit.event.Logout;
+import org.apache.logging.log4j.audit.LogEventFactory;
+import org.apache.logging.log4j.core.util.NetUtils;
+import org.acegisecurity.userdetails.UserDetails;
+
+/**
+ * Listener notified of user login and logout events.
+ */
+@Extension
+public class UserLogListener extends SecurityListener {
+
+    private static final Logger LOGGER = Logger.getLogger(UserLogListener.class.getName());
+
+    /**
+     * Fired when a user was successfully authenticated using credentials. It could be password or any other credentials.
+     * This might be via the web UI, or via REST (using API token or Basic), or CLI (remoting, auth, ssh)
+     * or any other way plugins can propose.
+     *
+     * UserLogListener does not use this abstract class method, however, rather than leaving
+     * its implementation empty it is made to throw an exception if the method is ever called.
+     *
+     * @param details details of the newly authenticated user, such as name and groups.
+     * @throws UnsupportedOperationException if the method is called wrongly.
+     */
+    @Override
+    protected void authenticated(@Nonnull UserDetails details) {
+        throw new UnsupportedOperationException("This class does not use this method.");
+    }
+
+    /**
+     * Fired when a user has failed to log in.
+     * Would be called after {@link #failedToAuthenticate}.
+     *
+     * UserLogListener does not use this abstract class method, however, rather than leaving
+     * its implementation empty it is made to throw an exception if the method is ever called.
+     *
+     * @param username the user
+     * @throws UnsupportedOperationException if the method is called wrongly.
+     */
+    @Override
+    protected void failedToLogIn(@Nonnull String username) {
+        throw new UnsupportedOperationException("This class does not use this method.");
+    }
+
+    /**
+     * Fired when a user tried to authenticate but failed.
+     * UserLogListener does not use this abstract class method, however, rather than leaving
+     * its implementation empty it is made to throw an exception if the method is ever called.
+     *
+     * @param username the user
+     * @throws UnsupportedOperationException if the method is called wrongly.
+     * @see #authenticated
+     */
+    @Override
+    protected void failedToAuthenticate(@Nonnull String username) {
+        throw new UnsupportedOperationException("This class does not use this method.");
+    }
+
+    /**
+     * Fired when a user has logged in, event logged via Log4j-audit.
+     *
+     * @param username name or ID of the user who logged in.
+     */
+    @Override
+    protected void loggedIn(@Nonnull String username) {
+        String currentTime = Instant.now().toString();
+        Login login = LogEventFactory.getEvent(Login.class);
+
+        try {
+            RequestContext.setHostName(NetUtils.getLocalHostname());
```

Comment:
We don't have a hostName attribute anywhere. Can you remove this from the RequestContext?